### PR TITLE
Feature: Add Theming Options

### PR DESCRIPTION
**Pull request by Adamj1232**
### Description
This PR adds a `theme` prop to the onboard initialization.
```typescript
interface InitOptions {
...
  /**
   * Custom or predefined theme for Web3Onboard i.e. default, dark, Custom, etc.
   */
  theme?: Theme
}

type Theme = ThemingMap | BuiltInThemes | 'system'

type BuiltInThemes = 'default' | 'dark' | 'light'

type ThemingMap = {
  '--w3o-background-color'?: string
  '--w3o-foreground-color'?: string
  '--w3o-text-color'?: string
  '--w3o-border-color'?: string
  '--w3o-action-color'?: string
  '--w3o-border-radius'?: string
}
```

The theme variables are mapped at the component level to fit the designs defined by @gesquinca 

## Default:
Screen Shot 2023-01-20 at 3 57 27 PM
Screen Shot 2023-01-20 at 3 57 20 PM
Screen Shot 2023-01-20 at 3 57 04 PM

## Dark:
Screen Shot 2023-01-20 at 3 57 48 PM
Screen Shot 2023-01-20 at 3 58 04 PM
Screen Shot 2023-01-20 at 3 58 11 PM

## Light:
Screen Shot 2023-01-20 at 3 58 57 PM
Screen Shot 2023-01-20 at 3 58 49 PM
Screen Shot 2023-01-20 at 3 58 32 PM

### Checklist
- [x] The version field in `package.json` is incremented following [semantic versioning](https://semver.org/)
- [x] The box that allows repo maintainers to update this PR is checked
- [x] I tested locally to make sure this feature/fix works
- [x] I have run `yarn file-check`, `yarn type-check` & `yarn build` to confirm there are not any associated errors
- [x] This PR passes the Circle CI checks

### CODE REVIEW

File: packages/core/src/views/account-center/Maximized.svelte [hunk 1]
```diff
@@ -164,56 +166,41 @@
...
       --account-center-border-radius,
       var(--onboard-border-radius-3, var(--border-radius-3))
     );
     padding: 12px;
+    border-top: 1px solid var(--border-color);
   }
 
   .app-name {
-    font-weight: 700;
+    /* font-weight: 700; */
```
Comment (Adamj1232) [inline packages/core/src/views/account-center/Maximized.svelte:180]:
```suggestion
```

File: packages/core/src/views/account-center/Maximized.svelte [hunk 2]
```diff
@@ -229,16 +216,16 @@
   }
 
   .app-button {
+    /* font-family: var(--account-center-app-btn-font-family, inherit); */
```
Comment (Adamj1232) [inline packages/core/src/views/account-center/Maximized.svelte:219]:
```suggestion
    font-family: var(--account-center-app-btn-font-family, inherit);
```
this was a requested variables so I would imagine it is in use

File: packages/core/src/views/account-center/Minimized.svelte [hunk 3]
```diff
@@ -61,56 +61,61 @@
...
-    background: var(
-      --account-center-minimized-background,
-      var(--onboard-white, var(--white))
-    );
-    border: 1px solid
-      var(--account-center-border, var(--onboard-gray-100, transparent));
+    --background-color: var(--account-center-minimized-background, var(--w3o-background-color, white));
+    --text-color: var(--w3o-text-color, var(--gray-700));
+    --border-color: var(--account-center-border, var(--w3o-border-color, var(--onboard-gray-200, var(--gray-200))));
+    /* --action-color: var(--w3o-accent-color); */
```
Comment (Adamj1232) [inline packages/core/src/views/account-center/Minimized.svelte:67]: what about here?

File: packages/core/src/views/shared/CloseButton.svelte [hunk 4]
```diff
@@ -1,31 +1,52 @@
...
+    position: relative;
+    overflow: hidden;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    height: 2rem;
+    width: 2rem;
+    border-radius: 2rem;
+    cursor: pointer;
+    /* color: var(
```
Comment (Adamj1232) [inline packages/core/src/views/shared/CloseButton.svelte:16]: I think this var was a community request too

File: packages/core/src/views/account-center/Maximized.svelte (hunk 2 above)
Comment (gesquinca) [reply packages/core/src/views/account-center/Maximized.svelte:219]: I'm not sure about this one. I REALLY don't want it, but let me look.

File: packages/core/src/views/account-center/Minimized.svelte (hunk 3 above)
Comment (gesquinca) [reply packages/core/src/views/account-center/Minimized.svelte:67]: I think I can just remove

File: packages/core/src/views/account-center/WalletRow.svelte [hunk 5]
```diff
@@ -67,48 +67,61 @@
     padding: 0.25rem;
     margin-bottom: 0.25rem;
     width: 100%;
-    font-size: var(--onboard-font-size-5, var(--font-size-5));
-    line-height: var(--onboard-font-line-height-2, var(--font-line-height-2));
+    /* font-size: var(--onboard-font-size-5, var(--font-size-5)); */
```
Comment (Adamj1232) [inline packages/core/src/views/account-center/WalletRow.svelte:70]: @gesquinca thoughts about these?

File: packages/core/src/views/connect/Sidebar.svelte [hunk 6]
```diff
@@ -20,16 +20,18 @@
 
 <style>
   .sidebar {
+    --background-color: var(--onboard-connect-sidebar-background, white);
+    --text-color: var(--onboard-connect-sidebar-color, var(--gray-700));
+    --border-color: var(--onboard-connect-sidebar-border-color, var(--gray-200));
+    /* --action-color: #00f; */
```
Comment (Adamj1232) [inline packages/core/src/views/connect/Sidebar.svelte:26]: can this be removed?

File: docs/src/lib/components/ThemeCustomizer.svelte [hunk 7]
```diff
@@ -18,27 +18,60 @@
...
+        id: 42161,
+        token: 'ARB-ETH',
+        label: 'Arbitrum',
+        rpcUrl: 'https://rpc.ankr.com/arbitrum'
       }
     ],
     appMetadata: {
       name: 'Documentation',
-      icon: '<svg></svg>',
+      // icon: '<svg></svg>',
```
Comment (Adamj1232) [inline docs/src/lib/components/ThemeCustomizer.svelte:59]:
```suggestion
```

File: packages/core/src/views/account-center/WalletRow.svelte [hunk 8]
```diff
@@ -64,51 +64,64 @@
...
   .balance {
     margin-left: 0.5rem;
-    color: var(--onboard-gray-300, var(--gray-300));
     transition: color 150ms ease-in-out, background-color 150ms ease-in-out;
     overflow: hidden;
     white-space: nowrap;
     text-overflow: ellipsis;
     width: 7.25rem;
     text-align: end;
+
```
Comment (Adamj1232) [inline packages/core/src/views/account-center/WalletRow.svelte:123]:
```suggestion
```

File: packages/core/src/views/account-center/WalletRow.svelte [hunk 9]
```diff
@@ -117,19 +130,20 @@
     border-radius: 24px;
     transition: color 150ms ease-in-out, background-color 150ms ease-in-out;
     background-color: transparent;
-    color: var(--onboard-gray-400, var(--gray-400));
+
```
Comment (Adamj1232) [inline packages/core/src/views/account-center/WalletRow.svelte:133]:
```suggestion
```

File: packages/demo/src/App.svelte [hunk 10]
```diff
@@ -239,8 +235,8 @@
     // },
     appMetadata: {
       name: 'Blocknative',
-      icon: blocknativeIcon,
-      logo: blocknativeLogo,
+      // icon: blocknativeIcon,
```
Comment (Adamj1232) [inline packages/demo/src/App.svelte:238]: @gesquinca what are we thinking here for the demo icons? Remove?

File: packages/hw-common/src/account-select.ts [hunk 11]
```diff
@@ -95,8 +95,17 @@ const mountAccountSelect = (
...
         /* SHADOWS */
         --shadow-1: 0px 4px 12px rgba(0, 0, 0, 0.1);
-      }
+        
+        /* THEMING */
+        --account-select-text-color: var(--w3o-text-color, initial);
+        --account-select-background-color: var(--w3o-background-color, initial);
+        --account-select-accent-background-color: var(--w3o-accent-background-color, initial);
+        --account-select-primary-100: var(--w3o-accent-background-color, initial);
+        --account-select-gray-500: var(--w3o-secondary-text-color, initial);
```
Comment (Adamj1232) [inline packages/hw-common/src/account-select.ts:104]: @gesquinca a few un-updated theming vars

File: packages/transaction-preview/src/views/Maximized.svelte [hunk 12]
```diff
@@ -54,20 +54,19 @@
 
 <style>
   .maximized {
-    font-family: var(--onboard-font-family-normal, var(--font-family-normal));
     pointer-events: all;
-    backdrop-filter: blur(5px);
+    /* backdrop-filter: blur(5px); */
```
Comment (Adamj1232) [inline packages/transaction-preview/src/views/Maximized.svelte:58]:
```suggestion
```

File: packages/transaction-preview/src/views/Maximized.svelte [hunk 13]
```diff
@@ -54,20 +54,19 @@
...
     width: 100%;
     min-height: 3.5rem;
-    background: var(--onboard-gray-600, var(--gray-600));
+    /* background: inherit; */
```
Comment (Adamj1232) [inline packages/transaction-preview/src/views/Maximized.svelte:61]:
```suggestion
```

File: packages/demo/src/App.svelte (hunk 10 above)
Comment (gesquinca) [reply packages/demo/src/App.svelte:235]: yeah, remove so we can test the default design. It's nice to be able to uncomment these for testing custom icon/logo options. That's why I left them there.

Comment (Adamj1232) [reply packages/demo/src/App.svelte:235]: I leave commented out for boilerplate examples

File: packages/hw-common/src/account-select.ts (hunk 11 above)
Comment (gesquinca) [reply packages/hw-common/src/account-select.ts]: Just updated these plus underlying components

**Review by aaronbarnardsound — APPROVED** [review]
That's a big one! Nice work. Approved with just a question around a possible memory leak.

File: packages/core/src/themes.ts [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+    )
+  })
+}
+
+export const watchForSystemThemeChange = (): void => {
+  const systemThemeDark = window.matchMedia('(prefers-color-scheme: dark)')
+  systemThemeDark.matches
+    ? handleThemeChange(themes['dark'])
+    : handleThemeChange(themes['light'])
+  systemThemeDark.addEventListener('change', ({ matches }) => {
```
[the remark this answers is not in the document]
Comment (Adamj1232) [reply packages/core/src/themes.ts]: Fixed! 🎸

**Review by gesquinca — APPROVED** [review]

**Pull request merged**